logserver/src/test/java/alluxio/logserver/AlluxioLog4jSocketNodeTest.java

```diff
@@ -0,0 +1,139 @@
+
+  private byte[] mBuffer;
+  private ValidatingObjectInputStream mValidatingObjectInputStream;
+
+  /**
+   * Only test the class in java.lang which can be serialized. Boolean, Byte, Character, Double,
+   * Float, Integer, Long, Short, String
+   */
+  private List<Object> createPositiveObjectList() {
+    Logger logger = Logger.getLogger("a");
+    return ImmutableList.of(
+        new Hashtable<>(),
+        new LoggingEvent(
+            "fqnOfCategoryClass",
+            logger,
+            Level.DEBUG,
+            "message",
+            new Throwable()),
+        new LocationInfo("c", "b", "c", "d"),
+        new ThrowableInformation(new Throwable()),
+        true,
+        (byte) 0x11,
+        0.0,
+        1.0F,
+        1,
+        1L,
+        (short) 1,
+        "string"
+    );
+  }
+
+  private List<Object> createNegativeObjectList() {
+    Logger logger = Logger.getLogger("a");
+    return ImmutableList.of(
+        new LogEvent(
+            "category",
+            logger,
+            Level.DEBUG,
+            "message",
+            new Throwable()),
+        new LogLevel("label", 1),
+        new AdapterLogRecord(),
+        new Log4JLogRecord(),
+        new LogTableColumn("label"),
+        Level.DEBUG,
+        UtilLoggingLevel.INFO,
+        new Throwable(),
+        new LogLevelFormatException("message"),
+        new LogTableColumnFormatException("message"),
+        new PropertySetterException("string")
+    );
+  }
+
+  @Test
+  public void testSerializables() {
+    List<Object> serializables = createPositiveObjectList();
+    for (Object object : serializables) {
+      // no exception expected
+      serializeThenDeserializeObject(object);
+    }
+  }
+
+  @Test
+  public void testNonSerializables() {
+    List<Object> serializables = createNegativeObjectList();
+    for (Object object : serializables) {
+      // exception expected
+      serializeThenDeserializeObject(object);
+    }
+  }
+
+  private void serializeThenDeserializeObject(Object object) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream)) {
+      objectOutputStream.writeObject(object);
+      mBuffer = byteArrayOutputStream.toByteArray();
+    } catch (IOException exception) {
+      System.out.println(exception.getMessage());
+    }
+
+    try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(mBuffer)) {
+      mValidatingObjectInputStream = new ValidatingObjectInputStream(byteArrayInputStream);
+      setAcceptList(mValidatingObjectInputStream);
+    } catch (IOException exception) {
+      System.out.println(exception.getMessage());
+    }
+
+    try {
+      Object deserializedObject = mValidatingObjectInputStream.readObject();
+      System.out.println(deserializedObject.getClass());
+    } catch (IOException exception) {
+      System.out.println(exception.getMessage());
```

Comment:
The expected `InvalidClassException` is silently caught and suppressed by this catch statement. 

Printing the exception message to console is NOT the same as failing the test. You need to make assertions, so that the test framework knows when a test fails.